src/sentry/static/sentry/app/views/organizationEventsV2/transactionView/span.tsx

```diff
@@ -0,0 +1,648 @@
+
+  toggleSpanTree = () => {
+    this.setState(state => {
+      return {
+        showSpanTree: !state.showSpanTree,
+      };
+    });
+  };
+
+  toggleDisplayDetail = () => {
+    this.setState(state => {
+      return {
+        displayDetail: !state.displayDetail,
+      };
+    });
+  };
+
+  renderDetail = ({isVisible}: {isVisible: boolean}) => {
+    if (!this.state.displayDetail || !isVisible) {
+      return null;
+    }
+
+    const {span} = this.props;
+
+    return <SpanDetail span={span} />;
+  };
+
+  getBounds = () => {
+    const {span, generateBounds} = this.props;
+
+    return generateBounds({
+      startTimestamp: span.start_timestamp,
+      endTimestamp: span.timestamp,
+    });
+  };
+
+  renderSpanTreeToggler = ({left}: {left: number}) => {
+    const {numOfSpanChildren} = this.props;
+
+    const chevron = this.state.showSpanTree ? <ChevronOpen /> : <ChevronClosed />;
+
+    if (numOfSpanChildren <= 0) {
+      return null;
+    }
+
+    return (
+      <SpanTreeTogglerContainer style={{left: `${left}px`}}>
+        <SpanTreeToggler
+          isExpanded={this.state.showSpanTree}
+          onClick={event => {
+            event.stopPropagation();
+
+            this.toggleSpanTree();
+          }}
+        >
+          <span style={{marginRight: '2px', textAlign: 'center'}}>
+            <Count value={numOfSpanChildren} />
+          </span>
+          <div style={{marginRight: '2px', width: '5px', textAlign: 'right'}}>
+            {chevron}
+          </div>
+        </SpanTreeToggler>
+      </SpanTreeTogglerContainer>
+    );
+  };
+
+  renderTitle = ({warningText}: {warningText?: string} = {}) => {
+    const {span, treeDepth} = this.props;
+
+    const op = span.op ? <strong>{`${span.op} \u2014 `}</strong> : '';
+    const description = get(span, 'description', span.span_id);
+
+    const MARGIN_LEFT = 8;
+    const TOGGLE_BUTTON_MARGIN_RIGHT = 8;
+    const TOGGLE_BUTTON_MAX_WIDTH = 40;
+
+    const left =
+      treeDepth * (TOGGLE_BUTTON_MAX_WIDTH + TOGGLE_BUTTON_MARGIN_RIGHT) + MARGIN_LEFT;
+
+    return (
+      <SpanBarTitleContainer>
+        {this.renderSpanTreeToggler({left})}
+        <SpanBarTitle
+          style={{
+            left: `${left}px`,
+            width: '100%',
+          }}
+        >
+          <span>
+            {op}
+            {description}
+          </span>
+          {warningText && (
+            <Tooltip title={warningText}>
+              <span style={{marginLeft: '8px', lineHeight: 0, height: '15px'}}>
+                <WarningIcon />
+              </span>
+            </Tooltip>
+          )}
+        </SpanBarTitle>
+      </SpanBarTitleContainer>
+    );
+  };
+
+  renderSpanChildren = () => {
+    if (!this.state.showSpanTree) {
+      return null;
+    }
+
+    return this.props.renderedSpanChildren;
+  };
+
+  renderSpanRow = () => {
+    const {span, spanBarColour} = this.props;
+
+    const startTimestamp: number = span.start_timestamp;
```

Comment:
Why number? Don't we generally pass around dates as either a Date or string or a moment object.

Reply:
@lynnagara right now, they're unix timestamps with sub-second precisions.